### PR TITLE
Fix save payment method checkbox for Subscriptions payment method update

### LINKED ISSUE #904: Subscriptions payment method change doesn't work if the save payment checkbox is not checked
**Bug description**

When viewing the change payment for a subscription page, the "save payment method" checkbox is visible and, if not selected, causes issues with the payment method change.

**To reproduce**

1. Subscribe to a Subscriptions product
2. Go to My account > Subscriptions > View > Change payment
3. Inform a new card and don't check the "save payment method" checkbox, then click on the Change payment method button
4. You should get a message saying that the subscription was updated
5. Try to renew that subscription - it's going to be charged using the previous card and not the newly updated one

**Expected behavior**

The "save payment method" checkbox should not be rendered when dealing with Subscriptions products. Also, any new card that's used for a subscription should be added to the user tokens.

### PR DESCRIPTION
**Pull request by luizreis**
Fixes #904 

#### Changes proposed in this Pull Request

* Hide the "save payment method" checkbox when changing the payment method for a subscription
* Save new cards added during the payment method update for a subscription

#### Testing instructions

1. Subscribe to a Subscriptions product
2. Go to My account > Subscriptions > View > Change payment
3. Assert that the "save payment method" checkbox is not displayed
4. Inform a new card and click on the Change payment method button
5. You should get a message saying that the subscription was updated
6. Renew that subscription
7. Assert that the renewal was processed using the new payment method

-------------------

- [x] Added changelog entry (or does not apply)
- [x] Covered with tests (or have a good reason not to test in description ☝️)
- [ ] Tested on mobile (or does not apply)

### CODE REVIEW

File: tests/test-class-wc-payment-gateway-wcpay-subscriptions-process-payment.php [hunk 1]
```diff
@@ -337,6 +337,47 @@ public function test_saved_card_is_added_before_status_update() {
...
+			->willReturn( $this->token );
+
+		// Expect add token to order to be called, so it can be reused in renewals.
+		$this->mock_wcpay_gateway
+			->expects( $this->once() )
+			->method( 'add_token_to_order' )
+			->with(
+				$this->callback( $this->match_order_id( $order->get_id() ) ),
+				$this->token
+			);
```
[the remark this answers is not in the document]
Comment (vbelolapotkov) [reply tests/test-class-wc-payment-gateway-wcpay-subscriptions-process-payment.php:372]: I've played with it and realized we'll have to fix mocking of `add_token_to_order` in all other tests which is out of scope of the PR. So let's move it into own issue (`Good first issue`) if it makes sense to refactor at all.

**Review by vbelolapotkov — APPROVED** [review]
Hey @luizreis ! The solution works well and code looks good, thank you! 🥇
I've left one comment regarding mocks in the test, would be happy to hear your thoughts :)
But this is no-blocker so :shipit:  after resolving conflicts :)

File: tests/test-class-wc-payment-gateway-wcpay-subscriptions-process-payment.php (hunk 1 above)
[the remark this answers is not in the document]
Comment (luizreis) [reply tests/test-class-wc-payment-gateway-wcpay-subscriptions-process-payment.php:372]: Yeah, it is a bit strange to mock one of the class's methods to test another one. The main reason for this was to split the tests for `add_token_to_order` and test it separately from the checkout flow, as it has a few edge cases. This mock allows the `process_payment` tests to avoid testing the add token logic, but rather check if it was called.

There's also the case of the `test_saved_card_is_added_before_status_update` test. We need to assert that the token was added before the order status is updated (more details in https://github.com/Automattic/woocommerce-payments/pull/832/commits/2dfec7df497014a6cf0582315821a8114fb9bd32). I'm not sure how to assert that if we're not mocking it unless we hook into the `woocommerce_order_status_changed` and check if the token is already there.

What are your thoughts on this? In case you think this partial mock is not doing us any favors, I'm fine with removing it and testing the basic `add_token_to_order` case on those tests, as mocking it does seem like a bit of a code smell.

[the remark this answers is not in the document]
Comment (vbelolapotkov) [reply tests/test-class-wc-payment-gateway-wcpay-subscriptions-process-payment.php:372]: Thanks for sharing your thoughts @luizreis . I see your point, but it's hard to comment on edge cases you've metnioned without deeper dive into it. I've just noticed from the first look that the code doesn't look like a candidate for mocking. I might be completely wrong here though 🙂

> What are your thoughts on this? In case you think this partial mock is not doing us any favors, I'm fine with removing it and testing the basic add_token_to_order case on those tests, as mocking it does seem like a bit of a code smell.

As I said above IMO it doesn't worth to refactor it as part of this PR. Creating an issue for that with comments above as context would be enough IMO, but the final decision is yours ;)

[the remark this answers is not in the document]
Comment (luizreis) [reply tests/test-class-wc-payment-gateway-wcpay-subscriptions-process-payment.php:372]: I think it's worth considering, and I agree that it doesn't belong to this PR. So, I created https://github.com/Automattic/woocommerce-payments/issues/920 for us to consider it and possibly remove it in the future.

Thank you for suggesting this!
